gslib/tests/test_hmac.py

```diff
@@ -357,3 +403,168 @@ def test_hmac_fails_for_xml(self):
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(('Gcloud Storage Command: {} alpha storage'
+                       ' hmac create {} --project {} {}').format(
+                           os.path.join('fake_dir', 'bin',
+                                        'gcloud'), hmac._CREATE_COMMAND_FORMAT,
+                           project, service_account), info_lines)
+
+  @mock.patch.object(cloud_api_delegator.CloudApiDelegator,
+                     'DeleteHmacKey',
+                     autospec=True)
+  def test_shim_translates_delete_command(self, mock_delete_hmac_key):
+    fake_cloudsdk_dir = 'fake_dir'
+    project = 'test-project'
+    access_id = 'fake123456789'
+    mock_delete_hmac_key.return_value = (
+        apitools_messages.HmacKeysDeleteResponse())
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': fake_cloudsdk_dir,
+      }):
+        mock_log_handler = self.RunCommand(
+            'hmac',
+            args=['delete', '-p', project, access_id],
+            return_log_handler=True)
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(('Gcloud Storage Command: {} alpha storage'
+                       ' hmac delete --project {} {}').format(
+                           os.path.join('fake_dir', 'bin', 'gcloud'), project,
+                           access_id), info_lines)
+
+  @mock.patch.object(cloud_api_delegator.CloudApiDelegator,
+                     'GetHmacKey',
+                     autospec=True)
+  def test_shim_translates_get_commannd(self, mock_get_hmac_key):
+    fake_cloudsdk_dir = 'fake_dir'
+    project = 'test-project'
+    access_id = 'fake123456789'
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': fake_cloudsdk_dir,
+      }):
+        mock_log_handler = self.RunCommand(
+            'hmac',
+            args=['get', '-p', project, access_id],
+            return_log_handler=True)
+
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(('Gcloud Storage Command: {} alpha storage'
+                       ' hmac describe {} --project {} {}').format(
+                           os.path.join('fake_dir', 'bin', 'gcloud'),
+                           hmac._DESCRIBE_COMMAND_FORMAT, project, access_id),
+                      info_lines)
+
+  @mock.patch.object(cloud_api_delegator.CloudApiDelegator,
+                     'ListHmacKeys',
+                     autospec=True)
+  def test_shim_translates_hmac_list_command(self, mock_list_hmac_keys):
```

Comment:
Are you testing short list and long list together?

Reply:
yep

Reply:
I split this test into 2:

- `test_shim_translates_hmac_list_command_using_long_format`
- `test_shim_translates_hmac_list_command_using_short_format`